vip-portal/src/main/resources/vm/wrapper.vm

```diff
@@ -19,6 +19,23 @@ function error {
   echo [ ERROR - $D ] $* >&2
 }
 
+function function_singularity {
+    cat > singularity << 'EOF'
+          #!/bin/bash
+          /cvmfs/dirac.egi.eu/container/apptainer/bin/apptainer "$@"
+EOF
+}
+
+# Call the function to create the singularity script and make it executable
+function_singularity
+chmod +x singularity
+export APPTAINER_PULLFOLDER="/cvmfs/dirac.egi.eu/container/apptainer/"
+
+# Define a function to run singularity commands
+singularity() {
+    ./singularity "$@"
+}
```

Comment:
It is better to add `PWD` to the `$PATH` like it's done for udocker

Reply:
done.